### ISSUE TITLE
Scheduled orchestration has different operation_id - correlation is not possible

### ISSUE THREAD

**Issue by s-krawczyk** [Azure/azure-functions-durable-extension#620]
Hello,

today I have found that scheduled orchestration has different operation_Id than this one by which it was started. It result with really hard correlation of logs in Application Insights. Not sure what is wrong but it looks like that inside it does not use System.Diagnostics.Activity information but produces his own operation id. Can someone explain if this is issue or not? Instance id in my case is not enough. We are loosing big picture what happened between all services.

**Comment by cgillum** [comment]
This is a known issue. We need to add support for correlation to the Durable Task Framework to address this. Some work has been done, but we don't yet have a timeline in place for when we can have it finished.

**Comment by fume** [comment]
Hi, any news on this? We're encountering the same "issue".

thanks,

**Comment by cgillum** [comment]
A [PR](https://github.com/Azure/durabletask/pull/261) has been created which adds this base functionality, but unfortunately the PR is quite complex and will take a while to prioritize and get through. We don't currently have an ETA for this.

**Comment by fume** [comment]
Hi @cgillum , i understand that the PR is quite complex. Thanks anyway for the feedback.

Regards, F

**Comment by RaviPa** [comment]
Is the PR still open?

**Comment by cgillum** [comment]
Yes, it's still being worked on.

**Comment by Gogeta094** [comment]
Hey. So much time til last message. 
Is there any updates? 
It's really important feature for production environments to be able to monitor your durable app easily.
And I find out it a not very trivial task to implement this correlation by myself in my Durable Function app.

**Comment by cgillum** [comment]
Yes, this is still work in progress. In the meantime, one trick you can do is use a naming convention for your orchestration IDs such that they all start with a common prefix - e.g. prefix with the parent orchestration's instance ID. Your log statements could similarly use these prefixes. You could then write queries in App Insights that filter using these prefixes to get all logs for the scenario you described.

Here is an example of this strategy in C#:

```csharp
[FunctionName(nameof(MyQueueTrigger))]
public async Task MyQueueTrigger(
    [QueueTrigger("my-queue")] string data,
    ExecutionContext executionContext,
    ILogger logger)
{
    // use the function invocation ID as the orchestration instance ID
    string instanceId = executionContext.InvocationId;
    logger.LogInformation($"Started orchestration with instance ID = {instanceId}.");
    await client.StartNewAsync(nameof(MyOrchestration), instanceId, data);
}

[FunctionName(nameof(MyOrchestration))]
public async Task RunOrchestrator(
    [OrchestrationTrigger] DurableOrchestrationContext context,
    ILogger logger)
{
    logger.LogInformation($"{context.InstanceId}: Message 1");
    // Prefix the sub-orchestration ID with the ID of the parent for easier searching
    string subInstanceId = $"{instanceId}.{suffix}";
    context.CallSubOrchestrationAsync("MySubOrchestration", subInstanceId, args);
}

[FunctionName(nameof(MySubOrchestration))]
public async Task RunOrchestrator(
    [OrchestrationTrigger] DurableOrchestrationContext context,
    ILogger logger)
{
    logger.LogInformation($"{context.InstanceId}: Message 2");
    // Call activity function MyActivityFunction1().
    // Call activity function MyActivityFunction2().
}

[FunctionName(nameof(MyActivityFunction1))]
public async Task<PolicyData> MyActivityFunction1(
    [ActivityTrigger] DurableActivityContextBase context,
    ILogger logger)
{
    logger.LogInformation($"{context.InstanceId}: Message 3");
    // Do work.
}

[FunctionName(nameof(MyActivityFunction2))]
public async Task MyActivityFunction2 (
    [ActivityTrigger] DurableActivityContextBase context,
    ILogger logger)
{
    logger.LogInformation($"{context.InstanceId}: Message 4");
    // Do work.
}
```

**Comment by MedAnd** [comment]
@cgillum - any update on this feature? Would be great to have an end to end sample & documentation for how to correlate across HttpTrigger, Durable Functions + Activities, Service Bus messages etc?

Also if using .Net Core 3+,  should one use Application Insights libraries directly or the new OpenTelemetry support?

**Comment by YunchuWang** [comment]
Closing as resolved.

The original starter-to-orchestration `operation_Id` correlation gap was addressed by Azure/durabletask#261 and Azure/durabletask#422, then shipped in Durable Functions v2.4.0 through #1571. That implementation was superseded by storage-provider-agnostic Distributed Tracing V2 (#2411), which is now GA. The corresponding .NET isolated starter correlation gap was fixed by #2998.

Current tracing propagates the caller context through `ParentTraceContext`; immediate and delayed (`StartAt`) starts use the same tracing path. Current correlation tests also verify a shared Application Insights operation ID and the expected parent chain.

Distributed Tracing V2 must be explicitly enabled:

```json
"tracing": {
  "distributedTracingEnabled": true,
  "version": "V2"
}
```

Current setup and minimum-version guidance: https://learn.microsoft.com/azure/durable-task/durable-functions/durable-functions-diagnostics#distributed-tracing

#2457 reports the same symptom on older tracing paths and should be re-triaged against V2. If the mismatch still occurs with V2 enabled on a supported version, please follow up there with the extension/runtime versions and a minimal reproduction.
